server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -601,44 +601,86 @@ private static String mergePaths(String path, String field) {
     }
 
     /**
-     * Returns all the concrete indices.
+     * Returns all the concrete indices in the format of array of strings.
      */
     public String[] getConcreteAllIndices() {
+        return allIndices.toArray(String[]::new);
+    }
+
+    /**
+     * Returns all the concrete indices in the format of set of strings.
+     */
+    public Set<String> getConcreteAllIndicesList() {
         return allIndices;
     }
 
     /**
-     * Returns all the concrete indices that are not hidden.
+     * Returns all the concrete indices that are not hidden in the format of array of strings.
      */
     public String[] getConcreteVisibleIndices() {
+        return visibleIndices.toArray(String[]::new);
```

Comment:
I don't think we have any utility left for the existing methods which are returning as arrays.

We probably just need a single method for each case which returns it as a list:

```
public List<String> getConcreteVisibleIndices() {
        return visibleIndices;
    }
```

Reply:
I had asked this query in bug - https://github.com/opensearch-project/OpenSearch/issues/8647#issuecomment-2188770011

Reply form Andrew was "This class is exposed to all OpenSearch plugins." Based on this I introduced new APIs.

Reply:
Got it. Makes sense.

I'm wondering what should be the plan to mark the old APIs as deprecated so that we remove them entirely in a later release.

Reply:
Yeah, it's fair game to put the `@Deprecated` annotation on these APIs and then remove them on the `main` branch for the 3.0 release.

Reply:
Updated this PR for 3.0 release as suggested. Just changed the return type of existing APIs from String array to Collection types.

Reply:
Thanks @akolarkunnu Changes make sense for 3.0 now with a minor code comment. Rest of it LGTM!

Also, removed 'backport-2.x' label to avoid auto-backporting to 2.x - I assume we will do a separate set of manual changes altogether for 2.x